arrow-core-data/src/main/kotlin/arrow/typeclasses/Apply.kt

```diff
@@ -38,11 +38,10 @@ interface Apply<F> : Functor<F> {
    */
   fun <A, B> Kind<F, A>.ap(ff: Kind<F, (A) -> B>): Kind<F, B>
 
-  /**
-   * Lazy version of ap, useful for datatypes which can short circuit.
-   * This will at some point be removed for a compiler plugin
-   */
-  fun <A, B> Kind<F, A>.lazyAp(ff: () -> Kind<F, (A) -> B>): Kind<F, B> = ap(ff())
+  fun <A, B> Kind<F, A>.lazyAp(ff: Eval<Kind<F, (A) -> B>>): Eval<Kind<F, B>> = ff.map { this.ap(it) }
```

Comment:
Like discussed on Slack.

Since this function is now not _lazy_ anymore but instead uses `Eval` we should rename to `apEval` which is fine since `apLazy` has not been released yet. Amazing work @1Jajen1 👏 👏